src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -99,6 +105,28 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
             throw new Exception("Invalid return URL");
         }
 
+        [HttpGet]
+        public async Task<IActionResult> Logout(string logoutId)
```

Comment:
Do we want to validate that `logoutId` is not null, empty or white space here? (I believe `GetLogoutRequest` will return `null` in that case and `Redirect` will throw if passed a null or empty string.)

Also, do we definitely want/need to provide GET given that a GET shouldn't have side effects?

Reply:
Yeah I will guard against the logoutId for null or white space. The GET is required as the logout would not work without it. Usually you would want to add a log out confirmation page here. Currently for this application we don't need it.

Reply:
There is the possibility to call  ```_interaction.CreateLogoutContextAsync(); ``` to create a logout Id if one is not supplied. Unfortunately this can still return null if there is not user. I have seen this used when using an external login / logout approach.